### PR TITLE
Display blocks to maturity on channel detail view.

### LINKED ISSUE #525: Show days+hours for force closed channels in channel detail view
Atm for force closed channels we show this on the main channel page
screen shot 2018-08-13 at 1 02 03 pm

As the channel is no longer _active_ and can't be used for payments, we can probably blank out the can send/receive columns. 

When users click down in they get this: 
screen shot 2018-08-13 at 5 03 39 pm

As is the ID field is blank, so users are unable to lookup the txid of the closing transaction on chain. Note that this is different from the txid of the _funding_ transaction. We can get the _closing_ txid from the `ForceClosedChannel` proto: https://api.lightning.community/#forceclosedchannel

From the same proto, we can also obtain the "blocks until maturity". Atm we don't display any sort of time estimate to when the funds will be swept back into the wallet. On mainnet, this can be anxiety inducing as you don't know when the funds will be swept back into your wallet. We've received a ton of issues just related to users not quite understanding timelocks on `lnd`, which caused us to surface more info on the command line. 

To start with, we can either display the block of blocks (which at least counts down), or attempt to map the # of blocks to days+hours. One block is 10 mins, and a days worth of blocks is 144, so we can use that as a starting point from there. The goal here to give users this feedback in "dark mode", so we can set proper expectations w.r.t _when_ their funds will be restored.

### PR DESCRIPTION
**Pull request by valentinewallace**
Closes #525.

"Blocks to maturity" doesn't seem like the best copy. Any ideas for what would be better?

### CODE REVIEW

File: src/view/channel-detail.js [hunk 1]
```diff
@@ -42,6 +42,11 @@ const ChannelDetailView = ({ store, nav }) => (
             {store.selectedChannel.closingTxId}
           </DetailField>
         ) : null}
+        {store.selectedChannel.blocksTilMaturity ? (
+          <DetailField name="Blocks Until Maturity">
```
Comment (Roasbeef) [inline src/view/channel-detail.js]: Rather than showing blocks, I think it'd be more intuitive for end users to show days/hours. So we can convert the blocks to days/hours using 144 blocks in a day (6 each hour).

Comment (Roasbeef) [reply src/view/channel-detail.js]: As for the copy, maybe: "Time Til Available", "Time Til Recovery"?

Comment (valentinewallace) [reply src/view/channel-detail.js]: fixed

File: src/action/channel.js [hunk 2]
```diff
@@ -346,6 +347,12 @@ class ChannelAction {
     const channel = pc.find(c => c.channelPoint === channelPoint);
     if (channel) pc.splice(pc.indexOf(channel));
   }
+
+  _formatTimeTilMaturity(numBlocks) {
+    const days = Math.floor(numBlocks / (24 * 6));
+    const hours = Math.floor((numBlocks % (24 * 6)) / 6);
+    return `${days} days and ${hours} hours`;
+  }
```
Comment (tanx) [inline src/action/channel.js]: Probably better to move to helper.js for easier unit testing

Comment (valentinewallace) [reply src/action/channel.js]: fixed

File: src/action/channel.js [hunk 3]
```diff
@@ -160,6 +160,7 @@ class ChannelAction {
         limboBalance: pfcc.limboBalance,
         maturityHeight: pfcc.maturityHeight,
         blocksTilMaturity: pfcc.blocksTilMaturity,
+        timeTilAvailable: formatTimeTilMaturity(pfcc.blocksTilMaturity),
```
Comment (tanx) [inline src/action/channel.js]: Rename to `getTimeTilAvailable` ... we're getting the time from the blocks.

File: src/view/channel-detail.js [hunk 4]
```diff
@@ -42,6 +42,11 @@ const ChannelDetailView = ({ store, nav }) => (
             {store.selectedChannel.closingTxId}
           </DetailField>
         ) : null}
+        {store.selectedChannel.blocksTilMaturity ? (
```
Comment (tanx) [inline src/view/channel-detail.js]: This will always be falsy ... change to `store.selectedChannel.timeTilAvailable`

File: test/unit/helper.spec.js [hunk 5]
```diff
@@ -817,4 +817,12 @@ describe('Helpers Unit Tests', () => {
...
     });
   });
+
+  describe('formatTimeTilMaturity()', () => {
+    it('should format blocks to human-readable time', () => {
+      const numBlocks = 463;
+      const time = helpers.formatTimeTilMaturity(numBlocks);
+      expect(time, 'to equal', '3 days and 5 hours');
+    });
+  });
```
Comment (tanx) [inline test/unit/helper.spec.js]: Nice test. Maybe add test cases for edge values like `undefined` and `0`.

File: src/action/channel.js (hunk 3 above)
Comment (valentinewallace) [reply src/action/channel.js]: fixed

File: src/view/channel-detail.js (hunk 4 above)
Comment (valentinewallace) [reply src/view/channel-detail.js]: fixed

File: test/unit/helper.spec.js (hunk 5 above)
Comment (valentinewallace) [reply test/unit/helper.spec.js]: fixed

File: src/view/channel-detail.js [hunk 6]
```diff
@@ -42,6 +42,11 @@ const ChannelDetailView = ({ store, nav }) => (
             {store.selectedChannel.closingTxId}
           </DetailField>
         ) : null}
+        {store.selectedChannel.timeTilAvailable ? (
+          <DetailField name="Time Til Available">
+            {store.selectedChannel.timeTilAvailable}
+          </DetailField>
+        ) : null}
```
Comment (tanx) [inline src/view/channel-detail.js]: Move this below `statusLabel`. Seems more natural to reading the time til available after the status.

File: test/unit/helper.spec.js [hunk 7]
```diff
@@ -817,4 +817,29 @@ describe('Helpers Unit Tests', () => {
...
+  describe('getTimeTilAvailable()', () => {
+    it('should format blocks to human-readable time', () => {
+      const numBlocks = 463;
+      const time = helpers.getTimeTilAvailable(numBlocks);
+      expect(time, 'to equal', '3 days and 5 hours');
+    });
+
+    it('should show NaN for undefined', () => {
+      const time = helpers.getTimeTilAvailable(undefined);
+      expect(time, 'to equal', 'NaN days and NaN hours');
```
Comment (tanx) [inline test/unit/helper.spec.js]: We should probably throw an Error for `undefined`. This should only happen if the api changed and `blocksTilMaturity` isn't set as expected. And we wanna get notified in that case.

Comment (valentinewallace) [reply test/unit/helper.spec.js]: fixed

File: src/view/channel-detail.js (hunk 6 above)
Comment (valentinewallace) [reply src/view/channel-detail.js]: fixed

File: src/helper.js [hunk 8]
```diff
@@ -321,3 +321,21 @@ export const generateArc = (x, y, radius, startAngle, endAngle) => {
...
 };
+
+/**
+ * Convert a number of blocks to an amount of time in the format "X days and Y
+ * hours" assuming 10 minutes per block.
+ * @param  {number} numBlocks The number of blocks to convert.
+ * @return {string} The amount of time the blocks is equivalent to.
+ */
+export const getTimeTilAvailable = numBlocks => {
+  let num = Number(numBlocks);
```
Comment (tanx) [inline src/helper.js]: This casting should not be necessary since `blocks_til_maturity` is of type number.
